File: app/src/main/java/com/android/sellacha/fragment/TransactionFragment.java
package com.android.sellacha.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.android.sellacha.R;
import com.android.sellacha.adapter.TransactionsAdapter;
import com.android.sellacha.databinding.FragmentTransactionBinding;
import com.android.sellacha.api.model.transactionsDM;

import java.util.ArrayList;

public class TransactionFragment extends Fragment {
    FragmentTransactionBinding binding;
    TransactionsAdapter transactionsAdapter;
    ArrayList<transactionsDM> transactionsArr = new ArrayList<>();

    @Override

    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_transaction, container, false);
        transactionsArr.add(new transactionsDM(4.5, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.2, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(5.0, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.5, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.2, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(5.0, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.5, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.2, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(5.0, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.5, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.2, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(5.0, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.5, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.2, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(5.0, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.5, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.2, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(5.0, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.5, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(4.2, "Completed", "COD", "#45245", "#546565635656"));
        transactionsArr.add(new transactionsDM(5.0, "Completed", "COD", "#45245", "#546565635656"));


        binding.transactionsRc.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
        transactionsAdapter = new TransactionsAdapter(getContext(), transactionsArr);
        binding.transactionsRc.setAdapter(transactionsAdapter);
        return binding.getRoot();
    }
}

File: app/src/main/java/com/android/sellacha/Transaction/activity/TransactionFragment.kt
package com.android.sellacha.Transaction.activity

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.android.sellacha.R
import com.android.sellacha.Transaction.Model.ModelTransaction
import com.android.sellacha.Transaction.adapter.AdapterTransaction
import com.android.sellacha.databinding.FragmentTransactionBinding
import com.android.sellacha.helper.myToast
import com.android.sellacha.utils.AppProgressBar
import com.example.ehcf.sharedpreferences.SessionManager
import com.example.myrecyview.apiclient.ApiClient
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class TransactionFragment : Fragment() {
    lateinit var binding: FragmentTransactionBinding
    private lateinit var sessionManager: SessionManager
    val handler = Handler(Looper.getMainLooper())

    val authorization = "Bearer 5|MuaOrW4WznVlYnCgjGRcJlopUqd2RJztSKcjorLG"
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_transaction, container, false)
    }

    @SuppressLint("LogNotTimber")
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding = FragmentTransactionBinding.bind(view)
        sessionManager = SessionManager(requireContext())


        CoroutineScope(Dispatchers.IO).launch {
            Log.d("FetchContact89", "fetchContacts: coroutine start")


        }
        binding.imgSearch.setOnClickListener {
            if (binding.edtTransId.text.toString().isEmpty()){
                binding.edtTransId.error="Enter Transaction Id"
                binding.edtTransId.requestFocus()
                return@setOnClickListener
            }
            val transactionId=binding.edtTransId.text.toString().trim()
            apiCallTransactionByTransId(transactionId)
        }
        apiCallTransaction()

    }

    private fun apiCallTransaction() {
        AppProgressBar.showLoaderDialog(requireContext())

        ApiClient.apiService.getTransaction(sessionManager.authToken)
            .enqueue(object : Callback<ModelTransaction> {
                @SuppressLint("LogNotTimber")
                override fun onResponse(
                    call: Call<ModelTransaction>, response: Response<ModelTransaction>
                ) {
                    if (response.code() == 500) {
                        myToast(requireActivity(), "Server Error")
                        AppProgressBar.hideLoaderDialog()

                    } else if (response.body()!!.data.orders.data.isEmpty()) {
                        binding.transactionsRc.adapter =
                            activity?.let { AdapterTransaction(it, response.body()!!) }
                        binding.transactionsRc.adapter!!.notifyDataSetChanged()
                        myToast(requireActivity(), "No Transaction Found")
                        AppProgressBar.hideLoaderDialog()
                    } else {
                        binding.transactionsRc.adapter =
                            activity?.let { AdapterTransaction(it, response.body()!!) }
                        AppProgressBar.hideLoaderDialog()


                    }
                }

                override fun onFailure(call: Call<ModelTransaction>, t: Throwable) {
                    myToast(requireActivity(), "Something went wrong")
                    AppProgressBar.hideLoaderDialog()


                }

            })
    }
    private fun apiCallTransactionByTransId(transactionId: String) {
        AppProgressBar.showLoaderDialog(requireContext())

        ApiClient.apiService.getTransactionByTransId(sessionManager.authToken,transactionId)
            .enqueue(object : Callback<ModelTransaction> {
                @SuppressLint("LogNotTimber")
                override fun onResponse(
                    call: Call<ModelTransaction>, response: Response<ModelTransaction>
                ) {
                    if (response.code() == 500) {
                        myToast(requireActivity(), "Server Error")
                        AppProgressBar.hideLoaderDialog()

                    } else if (response.body()!!.data.orders.data.isEmpty()) {
                        binding.transactionsRc.adapter =
                            activity?.let { AdapterTransaction(it, response.body()!!) }
                        binding.transactionsRc.adapter!!.notifyDataSetChanged()
                        myToast(requireActivity(), "No Transaction Found")
                        AppProgressBar.hideLoaderDialog()
                    } else {
                        binding.transactionsRc.adapter =
                            activity?.let { AdapterTransaction(it, response.body()!!) }
                        AppProgressBar.hideLoaderDialog()


                    }
                }

                override fun onFailure(call: Call<ModelTransaction>, t: Throwable) {
                    myToast(requireActivity(), "Something went wrong")
                    AppProgressBar.hideLoaderDialog()


                }

            })
    }

}